web/docs/language/features.md

```diff
@@ -178,138 +178,249 @@ Currently entities can be accessed only in Operations (Queries & Actions), so ch
-Then, you can either use it directly, or you can use it via special `useQuery` React hook (provided by Wasp**) to make it reactive.
 
-On server, you can import it the same way as on client, and then you can call it directly.
+#### Using the Query
+To use the Query, you can import it from `@wasp` and call it directly. As mentioned, the usage is the same regardless of whether you're on the server or the client:
+```javascript
+import fetchAllTasks from '@wasp/queries/fetchAllTasks.js'
+import fetchFilteredTasks from '@wasp/queries/fetchFilteredTasks.js'
+
+// ...
 
-**NOTE**: Wasp will not stop you from importing NodeJS function directly on server, e.g. `import { getAllTasks } from "./foo.js"`, but you shouldn't do it, because it will import pure NodeJS function and not a query recognized by Wasp, so it will not get all the features of a Wasp query.
+const allTasks = await fetchAllTasks();
+const doneTasks = await fetchFilteredTasks({isDone: true})
+```
 
-#### useQuery
-`useQuery` hook provided by Wasp is actually just a thin wrapper for `useQuery` hook from [react-query](https://github.com/tannerlinsley/react-query).
+**NOTE**: Wasp will not stop you from importing a Query's NodeJS implementation from `./queries.js` and calling it directly. However, we advise against this, as you'll lose all the useful features a Wasp Query provides (e.g., entity injection).
 
-You can import it as `import { useQuery } from '@wasp/queries'`.
+#### The `useQuery` hook
+When using Queries on the client, you can make them reactive with the help of the `useQuery` hook.
+This hook comes bundled with Wasp and is a thin wrapper around the `useQuery` hook from [_react-query_](https://github.com/tannerlinsley/react-query).
 
-Wasp `useQuery` takes three args:
-- `queryFn`: client query function generated by Wasp based on query declaration, e.g. one you get by importing in JS like this: `import getTasks from '@wasp/queries/getTasks.js'`.
-- `queryFnArgs`
-- `config`: react-query `config`.
+Wasp's `useQuery` hook takes three arguments:
+- `queryFn` (required): A Wasp query declared in the previous step or, in other words, the client-side query function generated by Wasp based on a `query` declaration.
+- `queryFnArgs` (optional): The arguments object (payload) you wish to pass into the query. The query's NodeJS implementation will receive this object as its first positional argument.
+- `config` (optional): A _react-query_ `config` object.
 
-It behaves exactly the same as [useQuery from react-query](https://react-query.tanstack.com/docs/api#usequery), only it doesn't take the key, that is handled automatically instead.
+Wasp's `useQuery` hook behaves mostly the same as [_react-query_'s `useQuery` hook](https://react-query.tanstack.com/docs/api#usequery), the only difference being in not having to supply the key. Wasp handles the caching automatically under the hood.
```

Comment:
I would say that Wasp handles not caching, but assigning a key, automatically under the hood.